sway/input/seat.c

```diff
@@ -52,6 +90,24 @@ struct sway_seat *sway_seat_create(struct sway_input_manager *input,
 		return NULL;
 	}
 
+	// init the focus stack
+	wl_list_init(&seat->focus_stack);
+	list_t *containers = container_list_children(&root_container);
+	if (containers == NULL) {
+		wlr_seat_destroy(seat->wlr_seat);
+		free(seat);
+		return NULL;
+	}
+	for (int i = containers->length - 1; i >= 0; --i) {
+		swayc_t *con = containers->items[i];
+		seat_container_from_container(seat, con);
+	}
+	free(containers);
+
+	wl_signal_add(&root_container.sway_root->events.new_container,
+		&seat->new_container);
+	seat->new_container.notify = handle_new_container;
```

Comment:
Is it necessary for every container to be represented in every focus stack? If not found, can't we just go up the parents until we find a container that works?

Reply:
(or return null if nothing is focused)

Reply:
My reasoning is that if we're ever in a situation where a new container doesn't get focused right away, this can be helpful to determine focus inactive because it encodes in what order the containers appeared. It might not be so easy though.

Reply:
I suppose that's reasonable.